### PR TITLE
Core/Spell: Hand Of Reckoning no damage when taunting players pet

### LINKED ISSUE #9068: [Core / Spell] Hand Of Reckoning no damage when taunting players pet
Rev:https://github.com/TrinityCore/TrinityCore/commit/9a652b390a82edf897cfc3e46d508a3fedf0fa8f

Bug: The Paladin Spell Hand Of Reckoning should deal damage to  targets which can be taunted and are not currently targeting you . It currently works only for creatures that are not controlled by the player . That should not happen , the spell should also work for creatures that the player controls . When you cast the spell on a pet that the player controls the pet will only receives the taunt effect and not the damage .

Currently the code is only 2 lines in spelleffects.cpp

 if (m_spellInfo->Id == 62124)
        m_caster->CastSpell(unitTarget, 67485, true);

The first id is the taunt effect , the second one is the damage.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/104109-core-spell-hand-of-reckoning-no-damage-when-taunting-players-pet?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Aokromes**
**Target branch(es)**: 335/6x

**Issues addressed**: Closes #9068

**Tests performed**: (Does it build, tested in-game, etc)

**Known issues and TODO list**:

By Effec7, closes #9068

### CODE REVIEW

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -3094,14 +3094,14 @@ void Spell::EffectTaunt(SpellEffIndex /*effIndex*/)
...
-    if (!unitTarget || !unitTarget->CanHaveThreatList()
+    if (!unitTarget || (!unitTarget->CanHaveThreatList() && !unitTarget->IsPet())
         || unitTarget->GetVictim() == m_caster)
     {
         SendCastResult(SPELL_FAILED_DONT_REPORT);
         return;
     }
 
-    if (m_spellInfo->Id == 62124)
+    if (m_spellInfo->Id == 62124 && !unitTarget->ToCreature()->HasReactState(REACT_PASSIVE))
```
Comment (jackpoz) [inline src/server/game/Spells/SpellEffects.cpp]: is unitTarget always a Creature ?

Comment (Aokromes) [inline src/server/game/Spells/SpellEffects.cpp]: @Effec7 https://github.com/TrinityCore/TrinityCore/pull/17518#discussion_r69758953

**Comment by ccrs** [comment]
I think this should be changed into 
````c++
if (!unitTarget || !unitTarget->CanHaveThreatList() || !unitTarget->HasPetAI() || unitTarget->GetVictim() == m_caster)
````

and ````HasPetAI```` should check if well the unit has a PetAI

**Comment by ariel-** [comment]
Players are immune to taunt effects. Should only trigger the damage spell if hit result didn't miss (like the check added in 08e0390d9503d4dd2b43dbe3224a70f3e76dec8c for SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER)

**Comment by ariel-** [comment]
Forget about previous comment, this was for player pets :P

**Pull request merged**